src/extension.ts

```diff
@@ -4,187 +4,62 @@ import * as vscode from 'vscode';
-                            vscode.window.showWarningMessage("Invalid typescript configuration file: " + configFile);
-                        } else {
-                            fileNames = config.fileNames;
-                            compilerOpts = config.options;
-                        }
-                    }
-                } catch (error) {
-                    vscode.window.showWarningMessage("Configuration file can not be read: " + configFile);
-                }
-            } else {
-                vscode.window.showWarningMessage("Configuration file not found, some features will not work properly!");
-            }
-
-            this.languageServiceHost = new LanguageServiceHost(fileNames, compilerOpts);
-            this.languageService = ts.createLanguageService(this.languageServiceHost);
-        }
 
-
-        private getDirectoryPath(path) {
-            return path.substr(0, Math.max(this.getRootLength(path), path.lastIndexOf('/')));
-        }
-        private getRootLength(path) {
-            if (path.charCodeAt(0) === 47 /* slash */) {
-                if (path.charCodeAt(1) !== 47 /* slash */)
-                    return 1;
-                var p1 = path.indexOf("/", 2);
-                if (p1 < 0)
-                    return 2;
-                var p2 = path.indexOf("/", p1 + 1);
-                if (p2 < 0)
-                    return p1 + 1;
-                return p2 + 1;
-            }
-            if (path.charCodeAt(1) === 58 /* colon */) {
-                if (path.charCodeAt(2) === 47 /* slash */)
-                    return 3;
-                return 2;
-            }
-            // Per RFC 1738 'file' URI schema has the shape file://<host>/<path>
-            // if <host> is omitted then it is assumed that host value is 'localhost',
-            // however slash after the omitted <host> is not removed.
-            // file:///folder1/file1 - this is a correct URI
-            // file://folder2/file2 - this is an incorrect URI
-            if (path.lastIndexOf("file:///", 0) === 0) {
-                return "file:///".length;
-            }
-            var idx = path.indexOf("://");
-            if (idx !== -1) {
-                return idx + "://".length;
-            }
-            return 0;
-        }
         private getSourceFile(document: vscode.TextDocument) {
-            const fileName = document.fileName;
-            const fileText = document.getText();
-            this.languageServiceHost.updateSnapshot(fileName, fileText);
-            return this.languageService.getSourceFile(fileName);
+            return ts.createSourceFile(document.fileName, document.getText(), ts.ScriptTarget.Latest, true);
         }
 
         provideCodeLenses(document: TextDocument, token: CancellationToken): CodeLens[] | Thenable<CodeLens[]> {
             var sourceFile = this.getSourceFile(document);
-            const methods: ts.Node[] = [];
+            const names: ts.Node[] = [];
```

Comment:
Couldn't this also be built using: `executeCommand('vscode.executeDocumentSymbolProvider', document.uri)` rather than needing to talk to typescript directly?

I am using this in my GitLens extension, [see here](https://github.com/eamodio/vscode-gitlens/blob/master/src/gitCodeLensProvider.ts#L42)